### DISCUSSION TITLE
Tuya TS0601 unsupported Zigbee Manufacturer _TZE200_kds0pmmv

### DISCUSSION THREAD

**Discussion by tobse257** [Koenkk/zigbee2mqtt#14320] [Devices]
Hi,

i just got a new Thermostat radiator valve from Moes which seems to be from an unsupported Zigbee Manufacturer _TZE200_kds0pmmv

I got it working with a convertert, but would it be possible to add to supported?

Thanks
Tobias

**Comment by Koenkk** [comment]
> I got it working with a convertert,

Please paste the converter here.

**Reply by tobse257** [reply]
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
		zigbeeModel: ['TS0601'],
        model: 'TV02-Zigbee',
        vendor: 'TuYa',
        description: 'Thermostat radiator valve',
        fromZigbee: [fz.ignore_basic_report, fz.ignore_tuya_set_time, fz.tvtwo_thermostat],
        toZigbee: [tz.tvtwo_thermostat],
        exposes: [
            e.battery_low(), e.child_lock(), e.open_window(), e.open_window_temperature().withValueMin(0).withValueMax(30),
            e.comfort_temperature().withValueMin(0).withValueMax(30), e.eco_temperature().withValueMin(0).withValueMax(30),
            exposes.climate().withPreset(['auto', 'manual', 'holiday']).withLocalTemperatureCalibration(-5, 5, 0.1, ea.STATE_SET)
                .withLocalTemperature(ea.STATE).withSetpoint('current_heating_setpoint', 0, 30, 0.5, ea.STATE_SET),
            exposes.numeric('boost_timeset_countdown', ea.STATE_SET).withUnit('second').withDescription('Setting '+
                    'minimum 0 - maximum 465 seconds boost time. The boost (♨) function is activated. The remaining '+
                    'time for the function will be counted down in seconds ( 465 to 0 ).').withValueMin(0).withValueMax(465),
            exposes.binary('frost_protection', ea.STATE_SET, 'ON', 'OFF').withDescription('When Anti-Freezing function'+
                    ' is activated, the temperature in the house is kept at 8 °C, the device display "AF".press the '+
                    'pair button to cancel.'),
            exposes.binary('heating_stop', ea.STATE_SET, 'ON', 'OFF').withDescription('Battery life can be prolonged'+
                    ' by switching the heating off. To achieve this, the valve is closed fully. To activate the '+
                    'heating stop, the device display "HS", press the pair button to cancel.'),
            e.holiday_temperature(),
            exposes.text('holiday_start_stop', ea.STATE_SET).withDescription('The holiday mode( ⛱ ) will '+
                    'automatically start at the set time starting point and run the holiday temperature.'),
            // exposes.enum('working_day', ea.STATE_SET, ['0', '1', '2', '3']),
            exposes.composite('schedule')/* .withFeature(exposes.text('week_schedule_programming', ea.STATE_SET)) */
                .withDescription('week_schedule').withDescription('Auto Mode ⏱ - In this mode, '+
                    'the device executes a preset week programming temperature time and temperature. '),
            exposes.text('schedule_monday', ea.STATE),
            exposes.text('schedule_tuesday', ea.STATE),
            exposes.text('schedule_wednesday', ea.STATE),
            exposes.text('schedule_thursday', ea.STATE),
            exposes.text('schedule_friday', ea.STATE),
            exposes.text('schedule_saturday', ea.STATE),
            exposes.text('schedule_sunday', ea.STATE),
            exposes.binary('online', ea.STATE_SET, 'ON', 'OFF').withDescription('Is the device online'),
            exposes.numeric('error_status', ea.STATE).withDescription('Error status'),
        ],
    };
	
	module.exports = definition;

**Reply by Koenkk** [reply]
Thanks, added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)
